### ISSUE TITLE
False positive on regex literal match for template strings

### ISSUE THREAD

**Issue by hcomnetworkers** [SonarSource/SonarJS#881]
**I want to report a bug.**

**SonarJS version:** 3.3 (Build 5702)<br>
**SonarQube version:** 6.7 (Build 33306)<br>

**Rule key:** javascript:UnusedVariable

**Reproducer** 
```
const foo = '.';
const bar = new RegExp(`\\${foo}`); // False-positive here
```
